### PR TITLE
fix: 10731 Focus composer without showing keyboard when users go to chats

### PR DESCRIPTION
**Pull request by zzz0108**
### Fixed Issues

$ https://github.com/Expensify/App/issues/10731
PROPOSAL: https://github.com/Expensify/App/issues/10731#issuecomment-1255955313

### Tests

1. Go to any chats
2. Verify that the composer is focused and the virtual keyboard is not shown. After touching on the composer, the keyboard is shown

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above

### QA Steps

1. Go to any chats
2. Verify that the composer is focused and the virtual keyboard is not shown. After touching on the composer, the keyboard is shown

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/21312517/bba2ba17-ac52-45d8-b4e9-da208bccae7b

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/21312517/d711bbea-7d41-482f-9bc0-acb59c6d9d2b

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/21312517/6622495d-3ae6-4183-8b5b-90f82f9ed073

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/21312517/5ce40f7e-b1bc-4d09-b6c3-658d71211624

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/21312517/ad06fab3-7176-4257-aac0-4fcd21bfca10

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/21312517/65e18cda-fc5d-48c7-8ec0-bc3126066ce9

</details>

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
> I am not sure if this is the expected behavior. 
@0xmiroslav Can you add more context for this why you think this way?

**Comment by Santhosh-Sellavel** [comment]
@ntdiary @0xmiroslav bump

> > I am not sure if this is the expected behavior.
>  @0xmiroslav Can you add more context for this why you think this way?

**Comment by Santhosh-Sellavel** [comment]
@christianwen can you resolve conflicts, please?

**Comment by Santhosh-Sellavel** [comment]
@christianwen Sorry for the delay, can resolve conflicts please?

**Comment by Santhosh-Sellavel** [comment]
@christianwen 
 Bug: Keyboard gets dismissed automatically on mWeb iOS

https://github.com/Expensify/App/assets/85645967/6801946f-b7fe-4279-80a5-3ffa42f8c30f

**Comment by Santhosh-Sellavel** [comment]
Any update?

**Comment by Santhosh-Sellavel** [comment]
Bump @christianwen

**Comment by Santhosh-Sellavel** [comment]
>  @christianwen 
>  Bug: On tapping composer to  bring Keyboard, Keyboard gets dismissed automatically on mWeb iOS
> 
> https://github.com/Expensify/App/assets/85645967/6801946f-b7fe-4279-80a5-3ffa42f8c30f
> 
> 

@christianwen this happens on mWeb iOS

**Comment by Santhosh-Sellavel** [comment]
Any update @christianwen

**Comment by Santhosh-Sellavel** [comment]
Any update @christianwen ?

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx [hunk 1]
```diff
@@ -185,10 +170,6 @@ const debouncedBroadcastUserIsTyping = lodashDebounce((reportID: string) => {
 
 const willBlurTextInputOnTapOutside = willBlurTextInputOnTapOutsideFunc();
 
-// We want consistent auto focus behavior on input between native and mWeb so we have some auto focus management code that will
-// prevent auto focus on existing chat for mobile device
-const shouldFocusInputOnScreenFocus = canFocusInputOnScreenFocus();
-
```
Comment (Santhosh-Sellavel) [inline src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:191]: Why this was deleted?

**Comment by Santhosh-Sellavel** [comment]
We've been working along on/off this issue for a long time. I think there might be some changes that be made intentionally looking at the changes. We might end-up creating a regression(s) by changing the current behaviour. 
So Before moving forward let's get confirmation from the team to continue. Can you create a post comparing the current behaviour & new behaviour. On the issue, so lets get consensus here.

**Comment by Santhosh-Sellavel** [comment]
Can you rephrase 

What's the current behavior?
What's the new behavior?

**Comment by Santhosh-Sellavel** [comment]
Bump @christianwen

**Comment by Santhosh-Sellavel** [comment]
Bump @christianwen

**Comment by Santhosh-Sellavel** [comment]
Lint fails

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx [hunk 2]
```diff
@@ -678,7 +658,15 @@ function ComposerWithSuggestions(
...
         // We avoid doing this on native platforms since the software keyboard popping
         // open creates a jarring and broken UX.
-        if (!((willBlurTextInputOnTapOutside || shouldAutoFocus) && !isNextModalWillOpenRef.current && !modal?.isVisible && isFocused && (!!prevIsModalVisible || !prevIsFocused))) {
+        if (
+            !(
+                (willBlurTextInputOnTapOutside || shouldAutoFocus) &&
+                !isNextModalWillOpenRef.current &&
+                !modal?.isVisible &&
+                isFocused &&
+                (!!prevIsModalVisible || !prevIsFocused || shouldInitFocus.current)
```
Comment (Santhosh-Sellavel) [inline src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:667]: why would we need `shouldInitFocus.current` here?

**Comment by Santhosh-Sellavel** [comment]
@christianwen Test fails for mWeb

https://github.com/Expensify/App/assets/85645967/011befcf-37f4-4ba4-bfd0-bc6309dc5eb8

**Comment by Santhosh-Sellavel** [comment]
Also in iOS paste option is shown on tap while touching on the composer to bring focus


https://github.com/Expensify/App/assets/85645967/863acc7c-a5e6-4820-ba41-6b7f9396f1fc

**Comment by Santhosh-Sellavel** [comment]
Do you have a solution for that?

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx (hunk 2 above)
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:667]: btw @christianwen?

**Comment by Santhosh-Sellavel** [comment]
@christianwen MarkdownTextInput has been for a while now, not sure how tests passed for you last week [here](https://github.com/Expensify/App/pull/32711#issuecomment-2046663440) 

Can you check and update here ?

**Comment by Santhosh-Sellavel** [comment]
> Also in iOS paste option is shown on tap while touching on the composer to bring focus
> 
>  Screen.Recording.2024-04-16.at.11.45.32.PM.mov

What about this @christianwen?

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel I mean all automatic tests (lint, jest, ...). `inputMode: none` works well in other inputs but composer because of [this reason](https://github.com/Expensify/App/pull/32711#issuecomment-2060470830). I think we should create the separate issue for that and hold this one util `inputMode` is added to rn-live-markdown

Yeah, let's go along to log an issue for this mWeb. And hold this, if input mode is added on **rn-live-markdown** that should automatically fix this right?

**Comment by Santhosh-Sellavel** [comment]
@christianwen Please update!

**Comment by Santhosh-Sellavel** [comment]
@christianwen [This](https://github.com/Expensify/App/pull/32711#issuecomment-2097403162) is recording from mWeb iOS 

The issues still exist on **iOS Native** as described in https://github.com/Expensify/App/pull/32711#issuecomment-2059685232

> Also in iOS paste/autofill popup option is shown on tap while touching on the composer to bring focus

#### Behaviour on Main
https://github.com/Expensify/App/assets/85645967/40e426a8-2803-462a-963f-14de0faf2d45

#### Behaviour on this Branch
https://github.com/Expensify/App/assets/85645967/4cae9754-8d70-40de-b820-58ff31c92d1f

**Comment by Santhosh-Sellavel** [comment]
@iwiznia 

Just Reiterating what's the expectation here, 

> 1. Go to any chats
> 2. Verify that the composer is focused and the virtual keyboard is not shown. After touching on the composer, the keyboard is shown

ATM here we struck with two issues here
1. One for mWeb the solution doesn't work because 
> it's the bug of react-native-live-markdown, we don't have `inputMode` property in MarkdownTextInput
> https://github.com/Expensify/react-native-live-markdown/blob/main/src/MarkdownTextInput.web.tsx

Possible next steps: Create an issue for this and hold this PR or merge as native works

2. Paste/Autofill popup shows on iOS Platform on tap, more details [here](https://github.com/Expensify/App/pull/32711#issuecomment-2098841951) 

>But in this PR we try to make the input focus automatically, if users press on the focused input, the menu will be shown (the default behavior of iOS)

Not sure second one is blocking or not.

Let us know how we van proceed on this?

**Comment by iwiznia** [comment]
Honestly I am not sure, I think that I rather stick with the current behavior than have those 2 new issues you describe.

**Comment by iwiznia** [comment]
I don't know, let's take it to the issue to discuss. Can you post a summary there and tag me and Christinadobrzyn?

**Comment by DylanDylann** [comment]
@christianwen I will try to complete my review this week. During this period, please help to resolve all errors (8 failing now)

**Comment by DylanDylann** [comment]
@christianwen I want to confirm this point: we always focus on the main composer when going to the chat screen for both empty chats and chats that have a long history 

It would be perfect if you help to refer to which comment that confirms the above point.

Because the new expect will against [the previous PR](https://github.com/Expensify/App/pull/18699)

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx (hunk 1 above)
Comment (DylanDylann) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:204]: Same question. 

After rechecking. This is added https://github.com/Expensify/App/pull/18699 and now we change the expectation. So this should be removed to align with new expect

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx [hunk 3]
```diff
@@ -285,23 +263,22 @@ function ComposerWithSuggestions(
     const {shouldUseNarrowLayout} = useResponsiveLayout();
     const maxComposerLines = shouldUseNarrowLayout ? CONST.COMPOSER.MAX_LINES_SMALL_SCREEN : CONST.COMPOSER.MAX_LINES;
 
-    const parentReportAction = parentReportActions?.[parentReportActionID ?? '-1'];
-    const shouldAutoFocus =
-        !modal?.isVisible &&
-        Modal.areAllModalsHidden() &&
-        isFocused &&
```
Comment (DylanDylann) [inline src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:292]: @christianwen  Why do we remove `isFocused` condition? Do this change break any other feature? I am struggling to find the reason why we add this condition

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx (hunk 2 above)
Comment (DylanDylann) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:652]: @christianwen Bump on this one. If possible, please rewrite this condition for more readable

**Comment by DylanDylann** [comment]
Leave a comment: https://github.com/Expensify/App/pull/32711/files#r1758376449

**Comment by DylanDylann** [comment]
@christianwen The contextMenuHidden doesn't work if the chat already have draft message

https://github.com/user-attachments/assets/581d9cf7-e6c2-48ae-95c5-ef4f0db4c69d

File: src/components/Composer/index.native.tsx [hunk 4]
```diff
@@ -109,6 +112,15 @@ function Composer(
     const maxHeightStyle = useMemo(() => StyleUtils.getComposerMaxHeightStyle(maxLines, isComposerFullSize), [StyleUtils, isComposerFullSize, maxLines]);
     const composerStyle = useMemo(() => StyleSheet.flatten([style, textContainsOnlyEmojis ? styles.onlyEmojisTextLineHeight : {}]), [style, textContainsOnlyEmojis, styles]);
 
+    useEffect(() => {
+        if (!showSoftInputOnFocus) {
+            return;
+        }
+        setTimeout(() => {
```
Comment (DylanDylann) [inline src/components/Composer/index.native.tsx:119]: To avoid using setTimeout, I suggest using isKeyboardShown

```
if (!showSoftInputOnFocus || !isKeyboardShown) {
            return;
}
setContextMenuHidden(false);
```

@christianwen Please try this approach only after resolving [this bug](https://github.com/Expensify/App/pull/32711#issuecomment-2348360000)

**Comment by DylanDylann** [comment]
@christianwen The autoFocus and keyboard don't work on safari

https://github.com/user-attachments/assets/5972ab37-0f63-4507-92a6-c97e4f8701cb

**Comment by DylanDylann** [comment]
@christianwen Please ping me when all dones

**Comment by DylanDylann** [comment]
@christianwen Please upgrade the RN live markdown version

**Comment by DylanDylann** [comment]
@christianwen Also please resolve conflict and failure test

**Comment by DylanDylann** [comment]
@christianwen You mean that this perf test is invalid now

**Comment by DylanDylann** [comment]
How about this comment: https://github.com/Expensify/App/pull/32711/files#r1758443883

**Comment by DylanDylann** [comment]
@christianwen Bump on https://github.com/Expensify/App/issues/10731#issuecomment-2363751533

**Comment by DylanDylann** [comment]
@christianwen If that, should we update the perf test to align with new change

**Comment by DylanDylann** [comment]
Ahh see. Thanks

**Comment by DylanDylann** [comment]
@christianwen Safari Bug: Keyboard doesn't appear


https://github.com/user-attachments/assets/4f5a5b96-c335-42b6-b9d7-16fbd7088480

**Comment by DylanDylann** [comment]
BUG: The keyboard doesn't show on Android (only show if clicking two times)

https://github.com/user-attachments/assets/e875bd73-2d67-48f8-b5de-7c0f3b118ec0

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]

**Comment by DylanDylann** [comment]
Just a note:
1. Changed files ESLint check
This action failed because of https://expensify.slack.com/archives/C01GTK53T8Q/p1726169039084589. This PR is complicated, so I think we should migrate withOnyx to useOnyx in another PR
2. Reassure Performance Tests
As mentioned [above](https://github.com/Expensify/App/pull/32711#issuecomment-2363066863)

> Updated! For the perf issue, we shouldn't mind about it since this test is to compare the number of re-render when users interact with input in baseline (from main) with the current behavior (this PR). In this PR we try to open the keyboard when they focus on the composer -> it takes 2 times to re-render

This PR will change the current behavior on purpose

**Comment by DylanDylann** [comment]
@iwiznia All yours

**Review by iwiznia — APPROVED** [review]

**Comment by iwiznia** [comment]
Checks failed

**Comment by DylanDylann** [comment]
@iwiznia Please check this comment: https://github.com/Expensify/App/pull/32711#issuecomment-2374541170

**Comment by iwiznia** [comment]
See https://github.com/Expensify/App/pull/32711#issuecomment-2374541170

**Comment by iwiznia** [comment]
Going to revert the PR since it caused https://github.com/Expensify/App/issues/49977

**Comment by DylanDylann** [comment]
@christianwen Could you create a quick PR to address [this bug](https://github.com/Expensify/App/issues/49977)?

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx [hunk 5]
```diff
@@ -298,17 +276,13 @@ function ComposerWithSuggestions(
...
-        (shouldFocusInputOnScreenFocus || (isEmptyChat && !ReportActionsUtils.isTransactionThread(parentReportAction))) &&
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:306]: @rojiphil This PR is reverted

[the remark this answers is not in the document]
Comment (rojiphil) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:306]: Oh! Thanks for pointing this out. Didn't notice this. Let me dig a little deeper.
